RtAudio.cpp

```diff
@@ -4796,14 +4890,24 @@ bool RtApiWasapi::probeDeviceOpen( unsigned int device, StreamMode mode, unsigne
       goto Exit;
     }
 
-    hr = devicePtr->Activate( __uuidof( IAudioClient ), CLSCTX_ALL,
-                              NULL, ( void** ) &renderAudioClient );
+    if (FAILED(hr = devicePtr->Activate(__uuidof(IAudioClient3), CLSCTX_ALL, NULL, (void**)&renderAudioClient))) {
+        hr = devicePtr->Activate(__uuidof(IAudioClient), CLSCTX_ALL, NULL, (void**)&renderAudioClient);
+    }
+    else {
+        iaudioclient3_available = true;
+    }
     if ( FAILED( hr ) ) {
       errorText_ = "RtApiWasapi::probeDeviceOpen: Unable to retrieve render device audio client.";
       goto Exit;
     }
 
-    hr = renderAudioClient->GetMixFormat( &deviceFormat );
+    if(iaudioclient3_available){
+        IAudioClient3* client3 = (IAudioClient3*)renderAudioClient;
+        hr = client3->GetCurrentSharedModeEnginePeriod(&deviceFormat,bufferSize);
```

Comment:
This will make the OpenStream function's parameter updated to be the system mixer buffer size. Perhaps a flag somewhere to disable the negotiation and always respect the user's input can be used.